app-web/plugins/gatsby-source-github-all/utils/fetchSourceGithub.js

```diff
@@ -224,14 +224,33 @@ const applyBaseMetadata = (
   };
 };
 
+/**
+ * filters files by the context directories from source
+ * @param {Array} files the files
+ * @param {Object} contextDir array of paths to get files in a repo
+ */
+const filterFilesByContext = (files, contextDir) => {
+  const targetPaths = contextDir.map(dir => {
```

Comment:
at one point we need to validate contextDir from the registry's the sourceproperties as it is an uncontrolled value

Reply:
perhaps allows contextDir to have 1 of 2 states
- type check it as a string and then convert it into an array [string]
- type check that it is an array and do not continue filtering if it isn't an array

and also maybe we provide a default value for the contextDir in the event it doesn't exist?
(somewhere?)
```javascript
const filterFilesByContext = (files, contextDir = []) => { 
```

Reply:
right, implemented it to use option 1